### PR TITLE
Update scikit-learn to 0.22.2

### PR DESCRIPTION
**Pull request by desilinguist**
This PR:

- updates scikit-learn from 0.21.3 to 0.22.2 
- updates test data due to this update
- updates import statements because of scikit-learn's API refactoring
- fixes a bug in `logutils.py`. 

**Note**: The conda and PyPI packages for 0.22.2 have a strange name this time (0.22.2.post1). This is most likely because they needed to fix something _after_ they had already uploaded 0.22.2 to PyPI which does not allow re-uploading with the same version number.

### CODE REVIEW

**Comment by desilinguist** [comment]
@aoifecahill @bndgyawali @AVajpayeeJr @ananyaganesh this PR is now ready for review.

File: examples/Tutorial.ipynb [hunk 1]
```diff
@@ -519,46 +518,46 @@
...
-      "| versicolor |        0 |          [2] |           2 |       1.000 |    0.500 |       0.667 |\n",
+      "| versicolor |        0 |          [3] |           0 |       1.000 |    1.000 |       1.000 |\n",
       "+------------+----------+--------------+-------------+-------------+----------+-------------+\n",
-      "|  virginica |        0 |            0 |         [4] |       0.667 |    1.000 |       0.800 |\n",
+      "|  virginica |        0 |            0 |         [4] |       1.000 |    1.000 |       1.000 |\n",
       "+------------+----------+--------------+-------------+-------------+----------+-------------+\n",
       "(row = reference; column = predicted)\n",
-      "Accuracy = 0.8181818181818182\n",
-      "Objective Function Score (Test) = 0.8181818181818182\n",
+      "Accuracy = 1.0\n",
```
Comment (bndgyawali) [inline examples/Tutorial.ipynb]: i do not see any update in the dataset used, but the results differ a lot. did i miss something?

File: tests/other/cv_results/test_grid_search_cv_results_predict_gs_f0_LogisticRegression.results.json [hunk 2]
```diff
@@ -1 +1 @@
...
...8348348347, 0.48348348348348347, 0.48348348348348347, 0.48348348348348347], "mean_test_score": [0.5339950729172286, 0.5169870469271668, 0.5169870469271668, 0.5169870469271668, 0.5169870469271668], "std_test_score": [0.006015492034256594, 0.023892771037679585, 0.023892771037679585, 0.023892771037679585, 0.023892771037679585], "rank_test_score": [1, 2, 2, 2, 2]}, "experiment_name": "test_grid_search_cv_results_predict_gs", "train_set_name": "f0.jsonlines", "train_set_size": 1000, "test_set_name": "f0.jsonlines", "test_set_size": 1000, "featureset": "[\"train_f0.jsonlines_test_f0.jsonlines\"]", "featureset_name": "f0", "shuffle": false, "learner_name": "LogisticRegression", "task": "predict", "start_timestamp": "10 Mar 2020 13:10:55.350373", "version": "2.0", "feature_scaling": "none", "folds_file": "", "grid_search": true, "grid_objective": "f1_score_micro", "grid_search_folds": "3", "min_feature_count": 1, "cv_folds": "None", "using_folds_file": false, "save_cv_folds": false, "save_cv_models": false, "use_folds_file_for_grid_search": true, "stratified_folds": true, "scikit_learn_version": "0.22.2.post1", "end_timestamp": "10 Mar 2020 13:10:55.409852", "total_time": "0:00:00.059479"}
```
Comment (bndgyawali) [inline tests/other/cv_results/test_grid_search_cv_results_predict_gs_f0_LogisticRegression.results.json]: `0.5339950729172286` vs `0.5260022124425846` grid score changes.

File: tests/test_cv.py [hunk 3]
```diff
@@ -20,22 +20,19 @@
 
 import numpy as np
 
-from nose.tools import eq_, raises
+from nose.tools import assert_less, assert_greater, eq_, raises
+from numpy.testing import assert_almost_equal
```
[the remark this answers is not in the document]
Comment (desilinguist) [reply tests/test_cv.py]: The semantics are different in the two functions so I didn't want to break anything by changing that.

**Comment by desilinguist** [comment]
@bndgyawali there are a lot of bug fixes and changes in scikit-learn so all of the changes in the numbers are due to that.

File: skll/learner.py [hunk 4]
```diff
@@ -2412,7 +2413,7 @@ def cross_validate(self,
                 kfold = StratifiedKFold(n_splits=cv_folds)
                 cv_groups = None
             else:
-                kfold = KFold(n_splits=cv_folds, random_state=random_state)
+                kfold = KFold(n_splits=cv_folds)
```
Comment (aoifecahill) [inline skll/learner.py]: If we're no longer passing in the random state, how is it picking it up?

Comment (desilinguist) [reply skll/learner.py]: According to the [docs](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold), `random_state` is only used if the `shuffle` keyword argument for `Fold` is `True` (it's `False` by default). With v0.22, setting `random_state` raises a warning unless `shuffle` is `True` so I removed it. Just to confirm, I double-checked the [0.21.3 documentation](https://scikit-learn.org/0.21/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold) and `shuffle` was `False` by default then too.
